src/realm/parser/driver.cpp

```diff
@@ -677,6 +679,11 @@ Query StringOpsNode::visit(ParserDriver* drv)
                 return Query(std::unique_ptr<Expression>(new Compare<Contains>(std::move(right), std::move(left))));
             case CompareNode::LIKE:
                 return Query(std::unique_ptr<Expression>(new Compare<Like>(std::move(right), std::move(left))));
+            case CompareNode::TEXT: {
+                StringData val = right->get_mixed().get_string();
```

Comment:
Why can't we have a specialized comparator for full text, like we do for all the other expressions? I would have expected this to be kind of convertible to some expression and being run inside a query.

Reply:
`StringNodeFulltext` is such a specialized node.